packages/editor/src/core/types/components.ts

```diff
@@ -11,4 +14,8 @@ export type Components = {
   BottomToolbar?: React.ComponentType<BottomToolbarProps>;
   CellPluginMissing?: React.ComponentType<CellPluginMissingProps>;
   EditModeResizeHandle?: React.ComponentType<{ onClick: () => void }>;
+  Cell?: React.ComponentType<CellProps>;
+  Row?: React.ComponentType<RowProps>;
+  HTMLCell?: React.ComponentType<HTMLCellProps>;
+  HTMLRow?: React.ComponentType<HTMLRowProps>;
```

Comment:
@macrozone, having different components for read-only and editable mode makes lots of sense in our use case. In fact we started with just Cell/Row and began using HTML versions of those much later. We use those (in particular) for enriching the editor's UI.

Looking at how Cell and Row are currently implemented, changing the interface to props-driver is going to be a significant rewrite. Might also add some unwarranted overhead (mixing props and hooks).

Reply:
It should be clear which components are used just for the editor-ui (in edit mode) and which are used for displaying content (and ends up on the visitor's screen).

Its also clear to me that if you customize how the data is rendered in read-only mode that the same customization is used in edit-mode (however this can be achieved). So if I e.g. pass a custom cell with a red border to HTMLCell (in your suggested api), i see the red border both in readonly and in edit mode. (i also think the name HTMLCell is missleading, its maybe CellWrapper or CellOuter or something like that). I believe anything else violates the "Principle of Least Surprise".

Being able to customize how a cell behaves **only** in edit mode is a different topic in my opinion and should be dealt sepratly.

Maybe you should first narrow down your requirements to find the best possible api?

Reply:
@macrozone , okay, so maybe we can call it CellHOC and have it conform to the HOC pattern, i.e. return a thunk that will be used for both Cell and HTMLCell purposes.

Reply:
@avoskresensky I think two other approaches would be better:

- allow to customize a `Cell` Component that takes a `children` prop which you have to render. So the default would be (more or less) just a `div` with the default grid' classes
- allow to customize a function that creates className and/or style  props to the cell's most outer div of a cell

I would avoid a HOC pattern.

also in this example above i do not distinguish between html-cell and editor cells, i strongly believe there should be one and only one property to customize the rendering of a cell for both modes.

(Changing **the behavior** of the editor for editing is a different topic)